modules/gapi/samples/pipeline_modeling_tool/utils.hpp

```diff
@@ -91,6 +93,26 @@ typename duration_t::rep timestamp() {
     return duration_cast<duration_t>(now.time_since_epoch()).count();
 }
 
+#define RETHROW_WITH_MSG_IF_FAILED(expr, msg)      \
+    try {                                          \
+        expr;                                      \
+    } catch (const std::exception& e) {            \
+        std::stringstream ss;                      \
+        ss << msg << "\n caused by: " << e.what(); \
+        throw std::logic_error(ss.str());          \
+    }                                              \
+
+template <typename K, typename V>
+void intersectMapWith(std::map<K, V>& target, const std::map<K, V>& second) {
+    for (auto&& item : second) {
+        auto it = target.find(item.first);
+        if (it != target.end()) {
+            throw std::logic_error("Met already existing key: " + item.first);
```

Comment:
"Error: key X already exists in Y" is better IMO.

Reply:
Yes, but X & Y are nameless.

Reply:
X is the value of your `item.first`.

Y is the map you're merging into.

Reply:
This `Y` doesn't have a name. Done